analyses/20_single-cell/23_tf_pw.py

```diff
@@ -50,8 +51,55 @@
 adata_nsclc = sc.read_h5ad(path_adata_nsclc)
 
 # %%
-adatas = {"m": {}, "nsclc": {}}
+# Make pseudobulk
+pb = dc.get_pseudobulk(
+    adata_nsclc, sample_col="patient", groups_col="cell_type", mode="sum", min_cells=10, min_counts=1000
+)
+
+sc.pp.normalize_total(pb, target_sum=1e6)
+sc.pp.log1p(pb)
```

Comment:
@czackl, how did you normalize data for the spatial analysis before running decoupler?

Reply:
We used CPM normalization as implemented in the spacedeconv function "normalize()"